src/Analyser/TypeSpecifier.php

```diff
@@ -1012,6 +1012,16 @@ private function specifyTypesForConstantBinaryExpression(
 						}
 					}
 
+					if ($context->truthy() && $isSize->maybe()) {
+						// turn optional offsets non-optional
```

Comment:
I'm not sure about this piece of code.

1) it only makes sense for lists
2) Why only for `$isSize->maybe()`?